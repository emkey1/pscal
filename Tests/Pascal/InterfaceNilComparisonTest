program InterfaceNilComparisonTest;

type
  IRunner = interface
    procedure Run;
  end;

  TRunner = record
    procedure Run; virtual;
  end;

procedure TRunner.Run;
begin
  { no-op }
end;

var
  runner: ^TRunner;
  maybe: IRunner;
begin
  maybe := nil;

  if maybe = nil then
    writeln('initial eq nil ok')
  else
    writeln('initial eq nil fail');

  if maybe <> nil then
    writeln('initial ne nil fail')
  else
    writeln('initial ne nil ok');

  new(runner);
  maybe := IRunner(runner);

  if maybe <> nil then
    writeln('assigned ne nil ok')
  else
    writeln('assigned ne nil fail');

  if maybe = nil then
    writeln('assigned eq nil fail')
  else
    writeln('assigned eq nil ok');

  maybe := nil;

  if maybe = nil then
    writeln('reset eq nil ok')
  else
    writeln('reset eq nil fail');

  if maybe <> nil then
    writeln('reset ne nil fail')
  else
    writeln('reset ne nil ok');

  dispose(runner);
end.
