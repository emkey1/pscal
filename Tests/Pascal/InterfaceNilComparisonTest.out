initial eq nil ok
initial ne nil ok
assigned ne nil ok
assigned eq nil ok
reset eq nil ok
reset ne nil ok
